### PR TITLE
40 mongo schema

### PR DESCRIPTION
**Pull request by cchauche**
Here are the fruits of my labor today:

I spent the whole day creating schema validation for mongoDB.  I then decided to make a set a sample data that was internally consistent to test out the validation.  I think this should give people a really good idea of the data we are working with in the current configuration. 

I've added a folder called "schemas" which contains the schema validation for each collection in mongoDB.  Basically that list is the master list of every field currently in a collection.  The one exception is that "_id" is omitted since all mongo documents have to have "_id".

I have crated a mongodump file so other can load the sample data and work with that in their local mongo file.  The sample data will automatically load the validation and the indexes I created.  Instrucitons to download and load the dump file will be posted in slack

### CODE REVIEW

File: db/schemas/timesheets.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+db.createCollection('timesheets', {
+  validator: {
+    $jsonSchema: {
+      bsonType: 'object',
+      required: [
+        'employee_id',
```
Comment (Orenjiku) [inline db/schemas/timesheets.js:6]: need 
'weekHours': {type: Number, default: 0}

File: db/schemas/employees.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        'state': {
+          bsonType: 'string'
+        },
+        'country': {
+          bsonType: 'string'
+        },
+        'zipcode': {
+          bsonType: 'string'
+        },
+        'wage': {
```
Comment (Orenjiku) [inline db/schemas/employees.js:41]: missing
  phone: {type: Number, default: ''},
  email: {type: String, required: true, unique: true},

File: db/schemas/employees.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          bsonType: 'decimal'
+        },
+        'startDate': {
+          bsonType: 'date'
+        },
+        'username': {
+          bsonType: 'string'
+        },
```
Comment (Orenjiku) [inline db/schemas/employees.js:55]: We will be using email as the username, so we won't be needing this username property. I already discussed with Aharon who did the employee front-end. 
Should discuss in the standup make sure authentication group is on the same page.

File: db/schemas/employees.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        'state': {
+          bsonType: 'string'
+        },
+        'country': {
+          bsonType: 'string'
+        },
+        'zipcode': {
+          bsonType: 'string'
```
Comment (Orenjiku) [inline db/schemas/employees.js:39]: zipcode should be a number

File: db/schemas/tasks.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        },
+        'isComplete': {
+          bsonType: 'bool'
+        },
+        'completedAt': {
+          bsonType: ['date', 'string']
+        },
+        'completedEmployee': {
+          bsonType: 'string'
+        }
```
Comment (Orenjiku) [inline db/schemas/tasks.js:50]: Some discrepancy between API and here
API has employeeCreated and employeeCompleted 
vs 
createdEmployee and completedEmployee

**Review by Orenjiku — COMMENTED** [review]
Made some comments on the schemas that I've been working on: employee, timesheet, task

File: db/schemas/employees.js (hunk 4 above)
Comment (cchauche) [reply db/schemas/employees.js:39]: I believe its best practice to save zipcodes as a string. please see this stack overflow: https://stackoverflow.com/questions/893454/is-it-a-good-idea-to-use-an-integer-column-for-storing-us-zip-codes-in-a-databas#:~:text=Use%20a%20string%20with%20validation,is%20up%20to%208%20characters).

File: db/schemas/employees.js (hunk 2 above)
Comment (cchauche) [reply db/schemas/employees.js:41]: I'll add phone and e-mail.  Phone should be a string please see this stack overflow: https://stackoverflow.com/questions/8284647/how-to-store-phone-numbers-on-mysql-databases

File: db/schemas/tasks.js (hunk 5 above)
Comment (cchauche) [reply db/schemas/tasks.js:50]: I'lll change to match api thank you

File: db/schemas/employees.js (hunk 3 above)
Comment (cchauche) [reply db/schemas/employees.js:55]: Its in there per my last conversation with Theo and Tina. Waiting on confirmation from them that I can get rid of it.

File: db/schemas/timesheets.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      ],
+      properties: {
+        'employee_id': {
+          bsonType: 'objectId',
+        },
+        'weekStart': {
+          bsonType: 'date',
+        },
+        'weekEnd': {
+          bsonType: 'date',
```
Comment (Orenjiku) [inline db/schemas/timesheets.js]: weekStart, weekEnd change to string

Comment (cchauche) [reply db/schemas/timesheets.js]: Got it, comming in new commit

**Pull request merged**